projects/plugins/boost/app/assets/src/js/features/critical-css/lib/generate-critical-css.ts

```diff
@@ -1,37 +1,80 @@
-import { get } from 'svelte/store';
-import { criticalCssMeta } from './stores/critical-css-meta';
-import {
-	localCriticalCSSProgress,
-	saveCriticalCssChunk,
-	criticalCssFatalError,
-	storeGenerateError,
-	updateProvider,
-} from './stores/critical-css-state';
-import {
-	CriticalCssState,
-	Critical_CSS_Error_Type,
-	Provider,
-} from './stores/critical-css-state-types';
-import { JSONObject } from '$lib/stores/data-sync-client';
+import { CriticalCssErrorDetails, Provider } from './stores/critical-css-state-types';
 import { recordBoostEvent, TracksEventProperties } from '$lib/utils/analytics';
 import { castToNumber } from '$lib/utils/cast-to-number';
 import { logPreCriticalCSSGeneration } from '$lib/utils/console';
 import { isSameOrigin } from '$lib/utils/is-same-origin';
 import { loadCriticalCssLibrary } from '$lib/utils/load-critical-css-library';
 import { prepareAdminAjaxRequest } from '$lib/utils/make-admin-ajax-request';
-import type { Viewport } from '$lib/utils/types';
+import { standardizeError } from '$lib/utils/standardize-error';
+import { SuccessTargetError } from 'jetpack-boost-critical-css-gen';
+
+const viewports = [
+	{
+		// Phone
+		width: 414,
+		height: 896,
+	},
+	{
+		// Tablet
+		width: 1200,
+		height: 800,
+	},
+	{
+		// Desktop
+		width: 1920,
+		height: 1080,
+	},
+];
```

Comment:
Minor issue: In case we'd want these be configurable in the future, can we accept these as a property in a function to keep functions pure?

Reply:
Fair point. Fixed in 267aab7653 - which turns these into `defaultViewports`, and adds `viewports` as an argument passed down the function chain.